### PR TITLE
Enable alpha test when using vertex alpha in some cases

### PR DESCRIPTION
**Pull request by bghgary**
This change addresses some edge cases tested by a new test model in the glTF sample assets repo: https://github.com/KhronosGroup/glTF-Sample-Assets/pull/146.

The issue is that alpha test is not taking into account whether a mesh has vertex alpha or not.

Before:
image

After:
image

TODO
- [ ] Add test once new test model is merged

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Materials/material.ts [hunk 1]
```diff
@@ -1147,13 +1147,22 @@ export class Material implements IAnimatable, IClipPlanesHolder {
         return false;
     }
 
+    /**
+     * Specifies whether or not this material should be rendered in alpha test mode when used with the given mesh.
+     * @param mesh defines the mesh to check
+     * @returns a boolean specifying if an alpha test is needed.
+     */
+    public needAlphaTestingForMesh(mesh: AbstractMesh): boolean {
```
Comment (ryantrem) [inline packages/dev/core/src/Materials/material.ts:1155]: Why public? Are you expecting it to be used elsewhere?

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15504/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15504/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15504/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15504/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15504/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15504/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15504/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15504/merge#BCU1XR#0

File: packages/dev/core/src/Materials/material.ts (hunk 1 above)
Comment (bghgary) [reply packages/dev/core/src/Materials/material.ts:1155]: Just following the pattern. `needAlphaBlendingForMesh` is also public. If we do this, we probably should be calling this function from other places that are currently calling `needAlphaTesting`.

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15504/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/Materials/material.ts (hunk 1 above)
Comment (Popov72) [reply packages/dev/core/src/Materials/material.ts:1155]: Yes, all current calls to `needAlphaTesting` (~10 after quick inspection) should probably be replaced by calls to `needAlphaTestingForMesh`.

Comment (bghgary) [reply packages/dev/core/src/Materials/material.ts:1155]: I've updated the code to use the new function. Still draft until the sample model is merged so that I can add a test, but the code part should be good now.

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15504/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15504/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bghgary** [comment]
Closing in favor of #15573
